src/connection.cpp

```diff
@@ -1200,6 +1213,9 @@ namespace oxen::quic
             tls_creds{context->tls_creds},
             di{*this}
     {
+        conn_established_cb = (context->conn_established_cb) ? std::move(context->conn_established_cb) : nullptr;
+        conn_closed_cb = (context->conn_closed_cb) ? std::move(context->conn_closed_cb) : nullptr;
+
```

Comment:
Discussed offline, but just dropping a reminder comment here: these two callbacks are a little bit weird since they are dual-purposed for both incoming and outgoing; I think we should split them up with separate callbacks specifiable for incoming and outgoing connection establishment/closing.

Reply:
Summarizing off-band discussion to ensure I'm on the same page:

We have two asymmetries in our connection open/close callbacks:
1) Callbacks for outbounds (passed to `Endpoint::connect(...)`) are to be used once, but callbacks for inbounds (passed to `Endpoint::listen(...)`) should be used for all inbound connections to that address
2) Our callback types should be differentiated between inbound/outbound callbacks, which may be another use-case for opt?

I can easily do bullet 1 in this PR; bullet 2 is a good stand-alone PR?